native-apps/unity_ii_universallink/unity_project/Assets/Scripts/TestICPAgent.cs

```diff
@@ -0,0 +1,75 @@
+using UnityEngine;
+using UnityEngine.UI;
+using EdjCase.ICP.Agent.Agents;
+using EdjCase.ICP.Agent.Identities;
+using EdjCase.ICP.Candid.Models;
+
+namespace IC.GameKit
+{
+    public class TestICPAgent : MonoBehaviour
+    {
+        public string greetFrontend = "https://nytzv-wqaaa-aaaan-qixdq-cai.icp0.io/";
+        public string greetBackendCanister = "n7s7b-3iaaa-aaaan-qixda-cai";
+
+        Text mMyPrincipalText = null;
+        Button mGreetButton = null;
+        Ed25519Identity mEd25519Identity = null;
+        DelegationIdentity mDelegationIdentity = null;
+
+        public Ed25519Identity TestIdentity { get { return mEd25519Identity; } }
+
+        internal DelegationIdentity DelegationIdentity
+        {
+            get { return mDelegationIdentity; }
+            set
+            {
+                mDelegationIdentity = value;
+
+                if (mDelegationIdentity != null && mGreetButton != null)
+                {
+                    mGreetButton.interactable = true;
+                }
+            }
+        }
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            var go = GameObject.Find("My Princinpal");
+            mMyPrincipalText = go?.GetComponent<Text>();
+
+            var buttonGo = GameObject.Find("Button_Greet");
+            mGreetButton = buttonGo?.GetComponent<Button>();
+
+            mEd25519Identity = Ed25519Identity.Generate();
```

Comment:
Is this using the iOS device HSM?

Reply:
It's an API provided by ICP.NET https://github.com/BoomDAO/ICP.NET/blob/main/src/Agent/Identities/Ed25519Identity.cs

Reply:
This seems to delegate to bouncy castle which in turn does not seem to integrate with the HSM/TPM, IIUC. It would be nice to have an implementation of this that _does_ use the TPM...

Reply:
Thanks. I will pass this message along to the ICP.NET team!